src/status_im2/subs/chat/messages.cljs

```diff
@@ -156,7 +156,11 @@
  (fn [pin-messages _]
    (->> pin-messages
         vals
-        (sort-by :pinned-at <))))
+        (sort-by :pinned-at
```

Comment:
newly pinned messages don't have `pinned-at` property until the user navigates away and back to the page so we just sort by the pinned messages with this pinned-at  value

Reply:
I tried to understand this sorting change @J-Son89, but I couldn't :man_shrugging: The code only sorts subsequent elements in the list containing a non-nil pinned-at value. Is this really what you expected? It's weird to see sorting that doesn't sort anything *sometimes*.

```clojure
(defn pinned-at-comparator
  [a b]
  (if (and a b)
    <
    (if a b a)))

;; Nothing is sorted because there are elements with nil interspersed.
(let [pin-messages [{:id 2 :pinned-at nil}
                    {:id 5 :pinned-at 9}
                    {:id 4 :pinned-at 11}
                    {:id 3 :pinned-at nil}
                    {:id 1 :pinned-at 10}]]
  (assert (= pin-messages
             (sort-by :pinned-at
                      pinned-at-comparator
                      pin-messages))))

;; The list is sorted in ascending order because there are no elements with nil interspersed.
(assert (= [{:id 5 :pinned-at 9}
            {:id 1 :pinned-at 10}
            {:id 4 :pinned-at 11}
            {:id 2 :pinned-at nil}
            {:id 3 :pinned-at nil}]
           (sort-by :pinned-at
                    pinned-at-comparator
                    [{:id 4 :pinned-at 11}
                     {:id 1 :pinned-at 10}
                     {:id 5 :pinned-at 9}
                     {:id 2 :pinned-at nil}
                     {:id 3 :pinned-at nil}])))

```

Reply:
so yeah, ideally there is cleaner code but this actually does exactly what I want :D

Reply:
Cool, thanks for the info @J-Son89.